ui/src/app/views/admin/hooks-tasks/details/hooks-task.component.ts

```diff
@@ -0,0 +1,115 @@
+import { formatDate } from '@angular/common';
+import { Component } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { TranslateService } from '@ngx-translate/core';
+import { HookStatus, TaskExecution, WorkflowHookTask } from '../../../../model/workflow.hook.model';
+import { HookService } from '../../../../service/services.module';
+import { Column, ColumnType } from '../../../../shared/table/sorted-table.component';
+
+@Component({
+    selector: 'app-hooks-task',
+    templateUrl: './hooks-task.html',
+    styleUrls: ['./hooks-task.scss']
+})
+export class HooksTaskComponent {
+    codeMirrorConfig: any;
+    columns: Array<Column>;
+    task: WorkflowHookTask;
+    executions: Array<TaskExecution>;
+    selectedExecution: TaskExecution;
+    selectedExecutionBody: string;
+    loading: boolean;
+
+    constructor(
+        private _hookService: HookService,
+        private _translate: TranslateService,
+        private _route: ActivatedRoute
+    ) {
+        this.codeMirrorConfig = this.codeMirrorConfig = {
+            matchBrackets: true,
+            autoCloseBrackets: true,
+            mode: 'application/json',
+            lineWrapping: true,
+            autoRefresh: true,
+            readOnly: true
+        };
+
+        this.columns = [
+            <Column>{
+                type: ColumnType.HTML,
+                name: '',
+                selector: d => {
+                    if (d.status === HookStatus.DONE) {
+                        return '<i class="check green icon"></i>';
```

Comment:
I don't feel very comfortable with the fact that we write HTML in typescript. Maybe we can find a better way to do this kind of code ?